image/tools/lib/backend/s3.sh

```diff
@@ -1,21 +1,55 @@
-#required env vars: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY and AWS_S3_BUCKET_NAME
-#optional env vars: AWS_S3_BUCKET_SUFFIX
+function check_backup_enabled {
+    local result=$(oc get secret ${BACKEND_SECRET_NAME} -n default 2>&1 | wc -l)
+    if [[ "$result"  == "1" ]]; then
+        return 1
```

Comment:
Should this be the other way around, if `result` is `1` then return code is `0` as the secret was found, else it's an error code `1`?

I tried this with the 3scale MySQL changes and I needed to reverse these to make the encryption logic be skipped if the secret doesn't exist.

Reply:
It's not using the return code. `oc` always returns 0, even if the secret does not exist. So instead i'm taking the line count of the response. If the result is 1 (1 line) then it means no secret was found and only a line containing `Not found.` was returned. If the secret was found then at least two lines (header + data) are returned.

Reply:
what do you think to making this:
```
local result=$(oc get secret ${BACKEND_SECRET_NAME} -n default | grep ${BACKEND_SECRET_NAME | wc -l)
if [[ "$result" == "0" ]]; then
    return 1;
fi
```
This is future-proofed against the output changing in the future (e.g. the header gets removed, or the not found error becomes multiline)

Reply:
@philbrookes this looks good. I'll give it a try and then change it to your version.

Reply:
@pb82 I meant what the return code from this function should be based on the line count from `oc`. Looking further into this, I'm seeing 2 lines when a secret isn't found when running on a bastion:

```
No resources found.
Error from server (NotFound): secrets "doesnt-exist" not found
```

Reply:
@aidenkeating @philbrookes I have updated those checks: oc will only return the secret name if it exists (using -o template). This has the advantage that it also works if `BACKEND_SECRET_NAME` is undefined as it does not rely on grep.